app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -5,17 +5,13 @@ module HasMonitoringManagerMixin
 
   def ensure_monitoring_manager
     # monitoring_manager should be defined by child classes.
-    if try(:monitoring_manager_needed?)
+    if try(:monitoring_manager_needed?) && monitoring_manager.nil?
       build_monitoring_manager(:parent_manager => self)
       monitoring_manager.name = "#{name} Monitoring Manager"
-    end
-    ensure_monitoring_manager_properties
-  end
-
-  def ensure_monitoring_manager_properties
-    if monitoring_manager
       monitoring_manager.zone_id = zone_id
       monitoring_manager.provider_region = provider_region
+    elsif try(:monitoring_manager_needed?) == false && monitoring_manager.present?
+      monitoring_manager.try(:delete_queue) unless try(:monitoring_manager_needed?)
```

Comment:
I don't think you need the `unless` here based on the condition above, right?

Reply:
Youre right. I was moving stuff around with @moolitayer